presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -583,6 +663,139 @@ public Void visitCall(CallExpression call, Context context)
                 return null;
             }
 
+            private boolean recursivelyExtractSubfieldsFromArrayLambdas(
+                    CallExpression rootCall,
+                    RowExpression expression,
+                    Context context,
+                    Set<Subfield> accessedSubfields,
+                    List<PathElement> prefixPath)
+            {
```

Comment:
this function is long and hard to read because of the nesting: can we refactor it?